lib/AIEDialect.cpp

```diff
@@ -177,6 +453,67 @@ AIEDialect::AIEDialect(mlir::MLIRContext *ctx)
 } // namespace AIE
 } // namespace xilinx
 
+// ObjectFifoCreateOp
+xilinx::AIE::TileOp xilinx::AIE::ObjectFifoCreateOp::getProducerTileOp() {
+  return cast<xilinx::AIE::TileOp>(producerTile().getDefiningOp());
+}
+
+xilinx::AIE::TileOp xilinx::AIE::ObjectFifoCreateOp::getConsumerTileOp() {
+  return cast<xilinx::AIE::TileOp>(consumerTile().getDefiningOp());
+}
+
+LogicalResult xilinx::AIE::ObjectFifoCreateOp::verify() {
+  if (size() < 2)
+    return emitError(
+        "Number of elements of AIE ObjectFifoCreateOp operation must be >= 2");
+
+  return success();
+}
+
+// ObjectFifoAcquireOp
+LogicalResult xilinx::AIE::ObjectFifoAcquireOp::verify() {
+  if (acqNumber() < 1)
+    return emitError("ObjectFifoAcquireOp must acquire at least one element");
+
+  if (port().getValue() != "produce" && port().getValue() != "consume")
```

Comment:
The port values, produce and consume, are currently used with the 'port' attribute instead of an input value (ex.: LockAction acquire/release). This was done as the relation between ObjectFifoAcquire(Release)Ops and port values is not as clear as that of LockOps and lock actions. Using the attribute 'port' allows to explicitly specify that relation.

Reply:
I don't quite understand.  IMHO, `AIE.objectFifo.acquire<Consume>(%objFifo, 2)` or `AIE.objectFifo.acquire<port=Consume>(%objFifo, 2)` is clearer than `AIE.objectFifo.acquire{ port = "consume">(%objFifo, 2)`, and it provides verifications in the TableGen, provides syntax documentation, and it is still explicitly specifying the relation.

Reply:
I see, I misunderstood what you meant. Indeed, the added support from TableGen will be helpful. I will make the change as soon as possible.

Reply:
Done.